common/flatpak-run.c

```diff
@@ -3805,10 +3886,17 @@ flatpak_run_app (FlatpakDecomposed *app_ref,
                             NULL);
 
   if (metakey != NULL &&
-      !flatpak_run_add_extension_args (bwrap, metakey, app_ref, use_ld_so_cache, &app_extensions, cancellable, error))
+      !flatpak_run_add_extension_args (bwrap, metakey, app_ref,
+                                       use_ld_so_cache, TRUE, NULL,
+                                       &app_extensions, cancellable, error))
     return FALSE;
 
-  if (!flatpak_run_add_extension_args (bwrap, runtime_metakey, runtime_ref, use_ld_so_cache, &runtime_extensions, cancellable, error))
+  if (!flatpak_run_add_extension_args (bwrap, runtime_metakey, runtime_ref,
+                                       use_ld_so_cache,
+                                       (runtime_files == original_runtime_files),
```

Comment:
I understand why you would not want to update LD_LIBRARY_PATH based on extensions here if the runtime is not on /usr. However, this looses information that would be needed if you were to try to use the extensions from /run/runtime. Maybe we need to propagate this info somewhere else? RUNTIME_LD_LIBRARY_PATH?

Reply:
Maybe we could write it into `/.flatpak-info`? It probably doesn't *need* to be an environment variable.

My intention was that pressure-vessel will build a modified `/usr` containing exactly the files it needs, where each file is either a hard link to a pristine Steam Runtime file, a symlink into `/run/host` or a symlink into `/run/runtime`. It will need to know the layout of the extensions while *building* that modified `/usr` before launching the container, at which time they are Steam's and pressure-vessel's `/usr`, but the launched game will not need to know that.

Reply:
True, you will need it before the .flatpak-info (or RUNTIME_LD_LIBRARY_PATH) is filled anyway, so maybe exposing this is not so important.

Reply:
I haven't taken action for this so far.